profile page assumes user is always found

This assumption leads to render unhandled errors

```
09:18:08.178 [io-executor-thread-1] ERROR org.thymeleaf.TemplateEngine - [THYMELEAF][io-executor-thread-1] Exception processing template "profile": Exception evaluating OGNL expression: "ghUser.avatarUrl" (template: "profile" - line 29, col 38)
org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating OGNL expression: "ghUser.avatarUrl" (template: "profile" - line 29, col 38)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:191)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:95)
        at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166)
        at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
        at org.thymeleaf.standard.expression.LinkExpression.executeLinkExpression(LinkExpression.java:264)
        at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:85)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
        at org.thymeleaf.standard.processor.AbstractStandardExpressionAttributeTagProcessor.doProcess(AbstractStandardExpressionAttributeTagProcessor.java:144)
        at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74)
        at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95)
        at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633)
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:918)
        at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228)
        at org.thymeleaf.engine.Model.process(Model.java:282)
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587)
        at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205)
        at org.thymeleaf.engine.Model.process(Model.java:282)
        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1587)
        at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205)
        at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
        at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592)
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1098)
        at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1067)
        at io.micronaut.views.thymeleaf.ThymeleafViewsRenderer.render(ThymeleafViewsRenderer.java:109)
        at io.micronaut.views.thymeleaf.ThymeleafViewsRenderer.lambda$render$1(ThymeleafViewsRenderer.java:96)
        at io.micronaut.core.io.Writable.writeTo(Writable.java:77)
        at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$encodeHttpResponse$11(RoutingInBoundHandler.java:1513)
        at io.micronaut.scheduling.instrument.InvocationInstrumenterWrappedRunnable.run(InvocationInstrumenterWrappedRunnable.java:47)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: ognl.OgnlException: source is null for getProperty(null, "avatarUrl")
        at org.thymeleaf.standard.expression.OGNLShortcutExpression.evaluate(OGNLShortcutExpression.java:89)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.executeExpression(OGNLVariableExpressionEvaluator.java:310)
        at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:170)
        ... 33 common frames omitted
```

--- a/.gitignore
+++ b/.gitignore
@@ -28,3 +28,5 @@ hs_err_pid*
 *.iml
 *.ipr
 *.iws
+
+.DS_Store

--- a/src/main/java/turbo/funicular/web/UiController.java
+++ b/src/main/java/turbo/funicular/web/UiController.java
@@ -71,7 +71,7 @@ public class UiController {
             .map(Principal::getName)
             .orElse("");
 
-        final var users = usersService.randomTop(5l)
+        final var users = usersService.randomTop(5L)
             .stream()
             .filter(user -> !user.getLogin().equals(username))
             .map(USERS_MAPPER::entityToCommand)

--- a/src/main/resources/i18n/messages.properties
+++ b/src/main/resources/i18n/messages.properties
@@ -1,6 +1,9 @@
 app.name=Funicular
 app.tagline=Let's share code using GitHub Gists!
 
+not-found.page.title=404 Not Found
+not-found.back.button.label=Back to Home Page
+
 home.page.title=User Home
 
 login.page.title=Log In to Funicular

--- a/src/main/resources/static/css/style.css
+++ b/src/main/resources/static/css/style.css
@@ -6,6 +6,10 @@
  * Made By MaterializeCSS.com
  */
 
+.sad {
+  width: 25%;
+}
+
 nav ul a,
 nav .brand-logo {
   color: #444;

--- a/src/main/resources/views/404.html
+++ b/src/main/resources/views/404.html
@@ -1,0 +1,31 @@
+<!DOCTYPE html>
+<html lang="en" th:replace="~{layoutFile :: layout(~{::title}, ~{::section})}" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <title th:text="#{not-found.page.title}"></title>
+</head>
+<body>
+<section>
+    <div class="container">
+        <div class="row">
+            <div class="col s12">
+                <h1 th:text="#{not-found.page.title}" class="header center-align"></h1>
+                <div class="card">
+                    <div class="card-image">
+                        <img src="https://media.giphy.com/media/QhCAwDXZ0BltK/giphy.gif"
+                             class="center-align">
+                    </div>
+                    <div class="card-stacked">
+                        <div class="card-content"></div>
+                        <div class="card-action">
+                            <a class="waves-effect waves-teal btn-flat btn-large"
+                               th:href="@{/home}"
+                               th:text="#{not-found.back.button.label}"></a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</section>
+</body>
+</html>

--- a/src/main/resources/views/profile.html
+++ b/src/main/resources/views/profile.html
@@ -4,7 +4,30 @@
     <title th:text="#{home.page.title}"></title>
 </head>
 <body>
-<section>
+<section th:if="${ghUser == null}">
+    <div class="container">
+        <div class="row">
+            <div class="col s12">
+                <h1 th:text="#{not-found.page.title}" class="header center-align"></h1>
+                <div class="card">
+                    <div class="card-image">
+                        <img src="https://media.giphy.com/media/QhCAwDXZ0BltK/giphy.gif"
+                             class="center-align">
+                    </div>
+                    <div class="card-stacked">
+                        <div class="card-content"></div>
+                        <div class="card-action">
+                            <a class="waves-effect waves-teal btn-flat btn-large"
+                               th:href="@{/home}"
+                               th:text="#{not-found.back.button.label}"></a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</section>
+<section th:unless="${ghUser == null}">
     <div class="container">
         <div class="row">
             <div class="col s6">
@@ -26,7 +49,7 @@
                     <div class="card-panel grey lighten-5 z-depth-1">
                         <div class="row valign-wrapper">
                             <div class="col s4">
-                                <img th:src="@{${ghUser.avatarUrl}}" alt="" class="circle responsive-img">
+                                <img th:src="@{${ghUser.avatarUrl}}" alt="" class="circle responsive-img" src="">
                             </div>
                             <div class="col s10">
                                 <h5 class="title" th:text="${ghUser.name}"></h5>

--- a/src/main/resources/views/user_home.html
+++ b/src/main/resources/views/user_home.html
@@ -47,7 +47,7 @@
                         <li class="collection-item avatar">
                             <img th:src="@{${user.avatarUrl}}" alt="" class="circle responsive-img">
                             <a class="title"
-                               th:href="@{/profile/{ghId}(ghId=${user.ghId})}"
+                               th:href="@{/profile/{login}(login=${user.login})}"
                                th:text="${user.name}"
                                ></a>
                             <p th:text="${user.bio}"></p>

